aya/src/programs/mod.rs

```diff
@@ -718,8 +720,12 @@ macro_rules! impl_fd {
         $(
             impl $struct_name {
                 /// Returns the file descriptor of this Program.
-                pub fn fd(&self) -> Option<ProgramFd> {
-                    self.data.fd.map(|fd| ProgramFd(fd))
+                pub fn fd(&self) -> Result<ProgramFd, ProgramError> {
+                    self.data.fd
+                        .as_ref()
+                        .ok_or(ProgramError::NotLoaded)
+                        .and_then(|fd| fd.try_clone().map_err(Into::into))
```

Comment:
Having the only way to get a the loaded file descriptor of a loaded program be through duplicating the file descriptor feels heavy weight to me. If we *really* don't want to spend much time thinking of another design at this time for how to handle borrowing two programs at the same time (or a program and a map at the same time) then we should at least provide a cheap way to get a borrowed file descriptor for the cases where the user can handle the lifetime themselves. In my use case I use the loaded programs file descriptors to setup tail calls and I normally remove the map out of the `Bpf` struct to handle their lifecycle myself so I wouldn't run into borrow checker issues but yet would still be paying the runtime penalty here.

Reply:
I've made `ProgramFd` return by reference. Does this appeal?

Reply:
Yeah that satisfies my concern enough. I am curious if we actually need the wrappers around the file descriptor at all but I guess that's tangential to this PR, they just feel kind of awkward and aren't providing that much value (preventing an user from trying to add a non-program fd to an extension/tail call table, which isn't unsafe it'd just return an error?).

Reply:
Shrug. I'd rather not engage on that discussion here.